core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/streams/streams.go

```diff
@@ -177,65 +176,97 @@ func (s *streams) buildResult(ctx context.Context, i int, checkResult ocr2keeper
 func (s *streams) doLookup(ctx context.Context, wg *sync.WaitGroup, lookup *mercury.StreamsLookup, i int, checkResults []ocr2keepers.CheckResult) {
 	defer wg.Done()
 
-	state, reason, values, retryable, retryInterval, err := mercury.NoPipelineError, mercury.MercuryUpkeepFailureReasonInvalidRevertDataInput, [][]byte{}, false, 0*time.Second, fmt.Errorf("invalid revert data input: feed param key %s, time param key %s, feeds %s", lookup.FeedParamKey, lookup.TimeParamKey, lookup.Feeds)
-	pluginRetryKey := generatePluginRetryKey(checkResults[i].WorkID, lookup.Block)
+	values, err := s.DoMercuryRequest(ctx, lookup, checkResults, i)
+	if err != nil {
+		s.lggr.Errorf("at block %d upkeep %s requested time %s DoMercuryRequest err: %s", lookup.Block, lookup.UpkeepId, lookup.Time, err.Error())
+	}
 
-	if lookup.IsMercuryV02() {
-		state, reason, values, retryable, retryInterval, err = s.v02Client.DoRequest(ctx, lookup, pluginRetryKey)
-	} else if lookup.IsMercuryV03() {
-		state, reason, values, retryable, retryInterval, err = s.v03Client.DoRequest(ctx, lookup, pluginRetryKey)
+	if err := s.CheckCallback(ctx, values, lookup, checkResults, i); err != nil {
+		s.lggr.Errorf("at block %d upkeep %s requested time %s CheckCallback err: %s", lookup.Block, lookup.UpkeepId, lookup.Time, err.Error())
```

Comment:
Hmm, dug further on this. There are also system errors, e.g. pack, unpack error. Are you saying we should just do Debugf here for all errors?

Reply:
I think `Errorf` might be appropriate here. I don't where a user issue would generate an error return from `CheckCallback()`. I only see packing/unpacking and rpc issues as potential sources of error.

Reply:
My thoughts on your log level question...

If the err only comes from user mistake --> use Debug()
If the err only comes from system problems --> use Error()
If the err could be the result of user or system problems --> we're logging in the wrong place and need more granularity

Reply:
Totally agreed with your analysis @RyanRHall. We are keeping this as Errorf since only errors come here.

> I don't where a user issue would generate an error return from CheckCallback(). I only see packing/unpacking and rpc issues as potential sources of error.

This is true after my latest update. here is the most recent commit: https://github.com/smartcontractkit/chainlink/pull/11489/commits/2eee0e9fca72ddd8a1846592eb1a70335b2ded7f